store/tikv/prewrite.go

```diff
@@ -169,21 +178,36 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 			c.setUndeterminedErr(errors.Trace(sender.rpcError))
 		}
 
+		// Unexpected error occurs, return it.
 		if err != nil {
 			return errors.Trace(err)
 		}
+
 		regionErr, err := resp.GetRegionError()
 		if err != nil {
 			return errors.Trace(err)
 		}
 		if regionErr != nil {
-			err = bo.Backoff(retry.BoRegionMiss, errors.New(regionErr.String()))
+			// For other region error and the fake region error, backoff because
+			// there's something wrong.
+			// For the real EpochNotMatch error, don't backoff.
+			if regionErr.GetEpochNotMatch() == nil || isFakeRegionError(regionErr) {
```

Comment:
These logics seem to duplicate in prewrite and commit... Is it possible to extract the logic? 
And does `BatchGet` also need it?

Reply:
> Is it possible to extract the logic?
I think it's not easy, because it has `continue` and a recursive function call.

> And does BatchGet also need it?
No, because only the backoff time of prewrite/commit is increased, retry logic of other requests doesn't change. But I'm thinking if we need to change more requests, for example, `PessimisticLock`.

Reply:
But I think it also possible for some read requests to trigger errors like region unavailable because it cannot read index during splitting...

Reply:
It will introduce many risks to do it for all read operations. I think the best solution is to eliminate the write stall, so that read won't be affected. The workaround for now is increasing the backoff time.